### ISSUE TITLE
Duplicated Symbols in Static Library Test Target

### ISSUE THREAD

**Issue by lawrencelomax** [CocoaPods/CocoaPods#1688]
Hi All,

I have an issue that relates to duplicated symbols in a static library and a test target. If I have a static library that has a dependency on ReactiveCocoa and then a dependency that depends on ```ReactiveCocoa``` in the test target, symbols will be duplicated in the static library and test target.

    pod 'ReactiveCocoa'
    
    target :MyLibTests, :exclusive => true do
      pod 'Specta'
      pod 'Expecta'
      pod 'OCMock'
      pod 'LLReactiveMatchers', :git => "https://github.com/lawrencelomax/LLReactiveMatchers.git"
    end

My static library requires ```ReactiveCocoa```, and therefore links it into the static library. ```MyLibTests``` requires ```LLReactiveMatchers```, which depends on ```ReactiveCocoa```. If there was a way of manually excludin  ```ReactiveCocoa``` from being linked with the test target, or if CocoaPods was capable if reasoning that ReactiveCocoa could be linked twice duplicate symbols could be prevented.

I suppose I could workaround this by making sure the test target does not link with the static library and manually adding each of my source files to the test target.

**Comment by jhollida24** [comment]
I'm trying to use LLReactiveMatchers and running in to this. My project uses ReactiveCocoa and the test target uses LLReactiveMatchers. Even if I use :exclusive => true on the test target, it gets linked twice.

**Comment by plivesey** [comment]
I believe this is still an issue. I created a new project (static library) and used this Podfile:
```
target 'CPTest', :exclusive => true do
  pod 'Isaac'
end

target 'CPTestTests', :exclusive => true do
  pod 'Isaac'
end
```

After running pod install, it warns me to add the inherited flag to other linker flags. Then I add an import to the tests and the library. After doing this, I get duplicate symbols when running tests. 

WORKAROUND: If I remove the inherited flag from tests, it works.

**Comment by AliSoftware** [comment]
Not sure, but this *may* be fixed by the future work on #3550 (WIP)

**Comment by segiddins** [comment]
Pretty sure this has been fixed by 0.38, can someone confirm?

**Comment by segiddins** [comment]
Closing due to lack of response.
